api/controllers/UserController.js

```diff
@@ -1,11 +1,15 @@
+import jwt from 'jsonwebtoken'
+
 module.exports = {
   create: function (req, res) {
-    const { name, email, password } = req.allParams()
+    const { name, email, email_validated, password } = req.allParams()
```

Comment:
It smells a little funny to me to have `email_validated` provided from the frontend. It would seem better to only. check if the code is validated on the code entry screen, then send the code again from the frontend when the user is actually created and remove the email validation record at the point of user creation. Obviously this would require a small set of amendments on both this back end and the front end implementations, but overall I think that would be more secure and potentially slightly simpler overall. I am anxious as I'm sure you are to get this off our plate, but if you're agree and are willing to be make the changes I can amend mobile quickly.

Reply:
yeah i could see that, i went this way because i actually thought maybe there is a future scenario where we want to allow people to create users without validated emails. so it was actually a choice to have the front-end handle the verification flow and tell us

Reply:
the one other advantage your version gives is that the code isnt deleted until the user is created so that the user can click the verify link multiple times and have it work (until it expires in 4 hours). This could be handy if a user clicks it on one device but then has to run off and then wants to verify on a different device shortly later. Not likely to happen a lot but maybe sometimes

Reply:
looking at it now, and feeling like this might be more of a refactor than i want to undertake right now. because i think it would mean changing how im handling the storing of emails (in the cookie)... honestly if its something you are excited to do id be happy to have you do it, otherwise id rather roll with what we have

Reply:
All things considered, I'm with you. It seems like it could linger a long time but I've went ahead and added a ticket to the backlog to review this again in the mythical quite moment down the road 😌 https://github.com/Hylozoic/hylo-node/issues/717